### ISSUE TITLE
Support proper error objects

### ISSUE THREAD

**Issue by jfhs** [Research-Institute/json-api-dotnet-core#63]
Specification [proposes](http://jsonapi.org/format/#error-objects) layout of `errors` part of the output. Currently available fields in `Internal\Error` are: title, detail and status. Specification has more fields, and library should allow user to set them.

Another issue is that it is currently formatter expects only single `Error` in response, it should also support `ErrorCollection`. Maybe a special exception `JsonApiMultipleErrors`, or something like that can be added, so that user can return multiple errors conveniently.

Ideally, there might be some support of validation (I don't know which one, does .Net core/EF has one?) and automatic exceptions if some fields are invalid thrown from Controller/DefaultRepository, with correct pointers to fields.

**Comment by jaredcnance** [comment]
A couple things I think we can do:

- Make `JsonApiExceptionFactory` a service so that it can be replaced by the users
- I think a special multi-exception is a good suggestion (or maybe just an overload for the current `JsonApiException`), its constructor might look like this. 

```
 public JsonApiMultiException(ErrorCollection errors);
```

The user's could customize the error object by inheriting from `Internal.Error` and adding properties as necessary. Then to throw the exception:

```
var myCustomErrors = new MyCustomErrors();
throw new JsonApiException(myCustomErrors);
```

What do you think?

**Comment by jaredcnance** [comment]
This is now available in the v1.2.0 pre-release. See the updated [documentation](https://github.com/Research-Institute/json-api-dotnet-core/tree/develop#custom-errors) for usage.
